utils/src/main/scala/za/co/absa/enceladus/utils/fs/S3FsUtils.scala

```diff
@@ -0,0 +1,271 @@
+
+    content
+  }
+
+  override def getDirectorySize(distPath: String): Long = getDirectorySize(distPath, _ => true)
+
+  /**
+   * Returns distributed directory size in bytes
+   */
+  private[fs] def getDirectorySize(distPath: String, keyNameFilter: String => Boolean): Long = {
+
+    // setup accumulation
+    val location = distPath.toS3Location(region)
+    val initSize = 0L
+
+    def accumulateSizeOp(previousTotalSize: Long, response: ListObjectsV2Response): Long = {
+      val objects = response.contents().asScala
+      val totalSize = objects
+        .filter(obj => keyNameFilter(obj.key))
+        .foldLeft(0L) { (currentSize: Long, nextObject: S3Object) => currentSize + nextObject.size }
+
+      previousTotalSize + totalSize
+    }
+
+    listAndAccumulateRecursively(location, accumulateSizeOp, initSize)
+  }
+
+  /**
+   * Hidden files = starting with `_` or `.` This method will return true for hidden keys.
+   *
+   * @param key path on s3
+   * @return e.g. `/path/to/.hidden` => true, `/path/to/non-hidden` => false
+   */
+  private[fs] def isKeyHidden(key: String): Boolean = {
+    val fn = key.split('/').last
+
+    (fn.startsWith("_")) || (fn.startsWith("."))
+  }
+
+  /**
+   * Returns distributed directory size in bytes, skipping hidden files and directories (starting from '_' or '.').
+   *
+   * @param distPath A path to a directory or a file.
+   * @return Directory size in bytes
+   */
+  override def getDirectorySizeNoHidden(distPath: String): Long = getDirectorySize(distPath, key => !isKeyHidden(key))
+
+
+  private[fs] def isKeyNonSplittable(key: String): Boolean = {
+    val fn = key.split('/').last
+
+    DistributedFsUtils.nonSplittableExtensions.exists(fn.endsWith)
+  }
+
+  /**
+   * Checks if the distributed-FS path contains non-splittable files
+   */
+  override def isNonSplittable(distPath: String): Boolean = {
+    // setup accumulation
+    val location = distPath.toS3Location(region)
+    val initFoundValue = false
+    // we want to break out of the recursion if a non-splittable is found, because it cannot ever be unfound.
+    val breakOutCase = Some(true)
+
+    def accumulateFoundOp(previouslyFound: Boolean, response: ListObjectsV2Response): Boolean = {
+      val objects = response.contents().asScala
+      val nonSplittableFound = objects.exists(obj => isKeyNonSplittable(obj.key))
+
+      previouslyFound || nonSplittableFound // true if ever found
+    }
+
+    listAndAccumulateRecursively(location, accumulateFoundOp, initFoundValue, breakOutCase)
+  }
+
+  /**
+   * Deletes a distributed-FS directory and all its contents recursively
+   */
+  override def deleteDirectoryRecursively(distPath: String): Unit = {
+
+    // setup accumulation
+    val location = distPath.toS3Location(region)
+
+    def accumulateSizeOp(acc: Unit, response: ListObjectsV2Response): Unit = { // side-effect, "accumulates" to unit
+      val objects = response.contents().asScala
+      if (objects.size > 0) {
```

Comment:
```suggestion
      if (objects.nonEmpty) {
```

Reply:
Not the only occurance

Reply:
Thanks, fixed.

Reply:
What about `keys.length > 0`

Reply:
True, I searched for `size > 0` 😆 . Fixed.